Modules/posixmodule.c

```diff
@@ -8588,6 +8588,49 @@ os_pidfd_open_impl(PyObject *module, pid_t pid, unsigned int flags)
 #endif
 
 
+#if defined(__linux__) && defined(HAVE_SETNS)
+/*[clinic input]
+os.setns
+  fd: fildes
+  nstype: int = 0
+
+Allows the calling thread to move into different namespaces.
+[clinic start generated code]*/
+
+static PyObject *
+os_setns_impl(PyObject *module, int fd, int nstype)
+/*[clinic end generated code: output=5dbd055bfb66ecd0 input=c097c9aa123c43ce]*/
+{
+    if (setns(fd, nstype) != 0) {
```

Comment:
We always release the GIL around syscalls unless we know for sure that the syscall cannot block, e.g. syscalls in vDSO.